### PR TITLE
feat: Add more info to the output APK filename

### PR DESCRIPTION
**Pull request by KobeW50**
This PR aims to add the app name and version to the name of the APK which is saved to external storage. For example, instead of the patched APK being named `base-ReX.apk`, it may now be named `ReX_YouTube_19.05.36.apk`.

I have no experience with shell, and I cannot read the code or test it. I imagine that my changes are not enough and will cause issues because I did not take reverse compatibility into account. Please carefully review (and test) the code if you are able to. If not, just close this PR.

Closes https://github.com/decipher3114/Revancify/issues/28

### CODE REVIEW

File: main.sh [hunk 1]
```diff
@@ -401,12 +401,12 @@ versionSelector() {
 }
 
 checkPatched() {
```
Comment (KobeW50) [inline main.sh:413]: I'm guessing some logic is needed in this function (lines 403-423) to check for the old file name format (and rm it if it exists) to allow reverse compatibility, but I'm not 100% sure

File: main.sh [hunk 2]
```diff
@@ -595,11 +595,11 @@ patchApp() {
...
+    java -jar "$cliSource"-cli-*.jar patch -fpw -b "$patchesSource"-patches-*.jar -m "$integrationsSource"-integrations-*.apk -o "apps/$appName-$appVer/$sourceName_$appName_$appVer.apk" $riplibArgs "${patchesArg[@]}" --keystore "$repoDir"/revancify.keystore --alias "decipher" --signer "decipher" --keystore-entry-password "revancify" --keystore-password "revancify" --custom-aapt2-binary ./aapt2 --options "$storagePath/$source-options.json" --exclusive "apps/$appName-$appVer/base.apk" 2>&1 | tee "$storagePath/patch_log.txt" | "${header[@]}" --begin 2 0 --ok-label "Continue" --cursor-off-label --programbox "Patching $appName $selectedVer.apk" -1 -1
     echo -e "\n\n\nRooted: $root\nArch: $arch\nApp: $appName v$appVer\nCLI: $(ls "$cliSource"-cli-*.jar)\nPatches: $(ls "$patchesSource"-patches-*.jar)\nIntegrations: $(ls "$integrationsSource"-integrations-*.apk)\nPatches argument: ${patchesArg[*]}" >>"$storagePath/patch_log.txt"
     tput civis
     sleep 1
-    if [ ! -f "apps/$appName-$appVer/base-$sourceName.apk" ]; then
+    if [ ! -f "apps/$appName-$appVer/$sourceName_$appName_$appVer.apk" ]; then
```
Comment (KobeW50) [inline main.sh:602]: I'm pretty sure an `or` operation is needed here to check for the old file name format.

**Comment by ChrisMiller94** [comment]
**Please check my opened issue here:**
https://github.com/decipher3114/Revancify/issues/88

It is very logical for naming, but unfortunately I'm not able to fix this.

@KobeW50 
Are you figured it out which things need to be replaced?

@decipher3114 
Please open a test or dev branch for testing purposes.

I'm really sure, you guys are able to think logical and fix my issue only with a few line code changes.

I'm able to test, if a test/dev branch will be available with a releated README for testing.

**Comment by KobeW50** [comment]
> **Please check my opened issue here:** #88
> 
> It is very logical for naming, but unfortunately I'm not able to fix this.
> 
> @KobeW50 Are you figured it out which things need to be replaced?

I agree that it would be a good change, but I am unfortunately not qualified to help. I can barely read code

**Pull request closed without merging**